### PR TITLE
DOCAPI-4293: DISTINCT ... ORDER BY, MySQL engine and mysql table function

### PR DESCRIPTION
**Pull request by BayoNet**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Category (leave one):
- Documentation

Updated the 'MySQL' engine descriprion.
Added the 'mysql' table function description.
Added description of special case, when the query contain DISTINCT and ORDER BY clases for different columns.

### CODE REVIEW

File: docs/ru/query_language/table_functions/mysql.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
+# mysql
+
+Allows to perform `SELECT` queries on data that is stored on a remote MySQL server.
```
Comment (alesapin) [inline docs/ru/query_language/table_functions/mysql.md]: Doesn't look like Russian language.

File: docs/en/query_language/select.md [hunk 2]
```diff
@@ -822,6 +822,43 @@ DISTINCT is not supported if SELECT has at least one array column.
...
+┌─a─┐
+│ 3 │
+│ 1 │
+│ 2 │
+└───┘
+```
+
+Row `2, 4` was cut before sorting.
+
+Take into account this implementation specificity when programming queries.
```
Comment (alesapin) [inline docs/en/query_language/select.md]: This is not `implementation specificity`. I think it's SQL standard behaviour.

**Review by alesapin — CHANGES_REQUESTED** [review]
.

File: docs/ru/query_language/table_functions/mysql.md (hunk 1 above)
Comment (BayoNet) [reply docs/ru/query_language/table_functions/mysql.md]: It's just a placeholder for the future Russian translation.

File: docs/en/query_language/select.md (hunk 2 above)
Comment (BayoNet) [reply docs/en/query_language/select.md]: As I have understood from the https://github.com/yandex/ClickHouse/issues/54 that is implementation specific behaviour.

**Review by alesapin — APPROVED** [review]

**Pull request merged**